### PR TITLE
Console overload for QSB

### PR DESCRIPTION
**Pull request by amazingalek**
* Console overload for QSB
* ~Deprecated ModConsole.OwmlConsole~

### CODE REVIEW

File: src/OWML.ModHelper.Menus/ModInputCombinationElementMenu.cs [hunk 1]
```diff
@@ -102,7 +100,7 @@ public void Initialize(PopupInputMenu menu)
 
 			if (layout == null)
 			{
-				ModConsole.OwmlConsole.WriteLine("Error - Failed to create combination visualizer in combination editor.", MessageType.Error);
+				Console.WriteLine("Error - Failed to create combination visualizer in combination editor.", MessageType.Error);
```
Comment (misternebula) [inline src/OWML.ModHelper.Menus/ModInputCombinationElementMenu.cs:103]: you've got unity code using system.console? or is this a variable named "Console"

File: src/OWML.Logging/ModConsole.cs [hunk 2]
```diff
@@ -5,6 +5,7 @@ namespace OWML.Logging
 {
 	public abstract class ModConsole : IModConsole
 	{
+		[Obsolete("Use ModHelper.Console instead.")]
 		public static ModConsole OwmlConsole { get; private set; }
```
Comment (misternebula) [inline src/OWML.Logging/ModConsole.cs:9]: i need this for QSB...

File: src/OWML.ModHelper.Menus/ModInputCombinationElementMenu.cs (hunk 1 above)
Comment (amazingalek) [reply src/OWML.ModHelper.Menus/ModInputCombinationElementMenu.cs:103]: it's just named Console. VS tells me it's correct, it's not the static Console built-in to C#

File: src/OWML.Logging/ModConsole.cs (hunk 2 above)
Comment (amazingalek) [reply src/OWML.Logging/ModConsole.cs:9]: Aye, removed it. Will think about it later.

File: src/OWML.Logging/OutputWriter.cs [hunk 3]
```diff
@@ -15,5 +15,8 @@ public class OutputWriter : IModConsole
 
 		public void WriteLine(string line, MessageType type) => 
 			ConsoleUtils.WriteByType(type, line);
+
+		public void WriteLine(string line, MessageType type, string senderType) => 
+			ConsoleUtils.WriteByType(type, $"{senderType}: {line}");
```
Comment (misternebula) [inline src/OWML.Logging/OutputWriter.cs:20]: this should be fixed as `{modName}.{senderType}` probably

Comment (misternebula) [reply src/OWML.Logging/OutputWriter.cs:20]: idk

Comment (amazingalek) [reply src/OWML.Logging/OutputWriter.cs:20]: actually this is only used by OWML in the launcher console , which means it's not used ;)
had to implement it as I added the method to the interface so mods (QSB) can access it easily.

File: src/OWML.Logging/ModSocketOutput.cs [hunk 4]
```diff
@@ -48,7 +48,9 @@ private void WriteLine(MessageType type, string line, string senderType)
 				KillProcess();
 			}
 		}
-
+		private void WriteLine(MessageType type, string line, string senderType) =>
+			WriteLine(line, type, senderType);
```
Comment (misternebula) [inline src/OWML.Logging/ModSocketOutput.cs:52]: does this not go to the `params object[] objects` method?

**Review by misternebula — APPROVED** [review]

**Pull request merged**